FileValidator.js

```diff
@@ -22,20 +25,27 @@ function FileValidator($input, dataAttrName, callback) {
     }
 
     $input.change(function() {
-        var validationCode = 'no_file_selected';
+        var validationCodes = ['no_file_selected'];
         if ($(this).val() !== '') {
-            validationCode = self.validate(this.files[0]);
+            validationCodes = self.validate(this.files[0]);
         }
 
-        callback.apply($input[0], [validationCode, $input[0].files[0], self.config]);
+        if (self.is_fetch_array) {
+            callback.apply($input[0], [validationCodes, $input[0].files[0], self.config]);
+        } else {
+            var validationCode = validationCodes.length ? validationCodes[0] : true;
+            callback.apply($input[0], [validationCode, $input[0].files[0], self.config]);
+        }
 
-        if (validationCode !== true) {
+        if (validationCodes.length) {
```

Comment:
Don't rely on truthy/falsy values if it's not necessary, use `validationCodes.length !== 0` instead.